### PR TITLE
Initial CDC support for DHC

### PR DESCRIPTION
**Pull request by jcferretti**
Deephaven - Google Chrome_140

### CODE REVIEW

File: docker-compose-common.yml [hunk 1]
```diff
@@ -14,10 +14,11 @@ services:
       # For remote debugging switch the line above for the one below (and also change the ports below)
       # - JAVA_TOOL_OPTIONS=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005 -Xmx4g -Ddeephaven.console.type=${DEEPHAVEN_CONSOLE_TYPE} -Ddeephaven.application.dir=${DEEPHAVEN_APPLICATION_DIR}
 
+
     expose:
       - '8080'
-#    ports:
-#      - '5005:5005'        # For remote debugging (change if using different port)
```
Comment (cpwright) [inline docker-compose-common.yml:20]: meant to be left open?

File: Integrations/python/deephaven/ConsumeCdc.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        KafkaConsumer specific desired configuration here.  Note this should include the relevant property
+        for a schema server URL where the key and/or value Avro necessary schemas are stored.
+    :param server_name:  The CDC server name.  This is used to compute the Kafka topic name (first component),
+        and also the Avro schema names expected in schema server for either key or value.
+    :param db_name:  The CDC database name.  This is used to compute the Kafka topic name (second component),
+        and also the Avro schema names expected in schema server for either key or value.
+    :param table_name:  The CDC table name.  This is used to compute the Kafka topic name (third component),
+        and also the Avro schema names expected in schema server for either key or value.
+    :param partitions: Either a sequence of integer partition numbers or the predefined constant
+        ALL_PARTITIONS for all partitions.  Defaults to ALL_PARTITIONS if unespecified.
```
Comment (cpwright) [inline Integrations/python/deephaven/ConsumeCdc.py:80]: s/unespecified/unspecified/

**Review by cpwright — COMMENTED** [review]
I think you've basically taken away a stream table as a valid output.  You can:
(1) ignore the keys and get a lastBy
(2) just make it append only, so you have a huge table
(3) what about if I know what I'm doing and want the raw stream table instead?  Like maybe I don't actually care about these individual things, but I want to aggregate them with counts and sums.  Or in an advanced use case maybe I can instead do something like:
purchases = streamTable.where(op = add).view("Dollars").sumBy()
returns = streamTable.where(op = del).view("Dollars").sumBy()
netRevenue = purchases.naturalJoin(returns, "", "ReturnDollars=Dollars").update("NetRevenue=Dollars-ReturnDollars")

**Comment by jcferretti** [comment]
Ref https://github.com/deephaven/deephaven-core/pull/1819#pullrequestreview-849885407
(I can't answer directly under that comment, sadly, very poor github UI design).
Makes sense and it becomes obvious when looking at the implementation code; just returning the streaming table before wrapping it in a call to streamingToAppendTable is obviously more general and the user themselves can do the wrapping if they so desire.

I've reworked some other parts of the change too; there was a bug in that the after__X columns are not present on a delete, so it is important to take the primary key columns always from the Key component (those are present in a delete).

Something else I learned from debezium documentation, when an update in the underlying database table modifies a primary key, debezium generates two events, one to delete the old row and one to add the new row.  Is very important for our algorithm that they do this, otherwise the trick for 

```
.where("op = `d`")
```

wouldn't work; I thought about this case,  went into panic, went to read the debezium docs, and was relieved.

File: extensions/kafka/src/main/java/io/deephaven/kafka/CdcTools.java [hunk 3]
```diff
@@ -2,99 +2,370 @@
...
-                        ? KafkaTools.Consume.IGNORE
-                        : KafkaTools.Consume.avroSpec(keySchema),
+                KafkaTools.Consume.avroSpec(keySchema),
                 KafkaTools.Consume.avroSpec(valueSchema),
                 KafkaTools.TableType.Stream);
         final List<String> dbTableColumnNames = dbTableColumnNames(streamingIn);
+        final String[] allDroppedColumns;
+        if (dropColumns != null && dropColumns.size() > 0) {
+            int i = 0;
+            allDroppedColumns = new String[dropColumns.size() + (asStreamTable ? 0 : 1)];
```
Comment (cpwright) [inline extensions/kafka/src/main/java/io/deephaven/kafka/CdcTools.java:307]: maybe just Arrays.copyOf?

**Review by cpwright — DISMISSED** [review]

**Review by cpwright — DISMISSED** [review]

**Review by cpwright — APPROVED** [review]

**Pull request merged**